src/jose/Jws.ts

```diff
@@ -0,0 +1,132 @@
+    const protectedHeaderBase64UrlString = base64url.encode(protectedHeaderString);
+    const payloadBase64UrlString = base64url.encode(payload);
+    const signingInputBase64urlString = protectedHeaderBase64UrlString + '.' + payloadBase64UrlString;
+    const signingInputBuffer = Buffer.from(signingInputBase64urlString);
+
+    // This is where we will add support for different algorithms over time.
+    let signatureBuffer: Buffer;
+    if (privateKeyJwk.crv === 'Ed25519') {
+      signatureBuffer = await Jws.signEd25519(signingInputBuffer, privateKeyJwk as JwkEd25519Private);
+    } else if (privateKeyJwk.crv === 'secp256k1') {
+      signatureBuffer = await Jws.signSecp256k1(signingInputBuffer, privateKeyJwk as JwkSecp256k1Private);
+    } else {
+      throw new Error(`unsupported key type ${privateKeyJwk.kty} with curve ${privateKeyJwk.crv}`);
+    }
+
+    const signatureBase64UrlString = base64url.encode(signatureBuffer);
+
+    return {
+      payload   : payloadBase64UrlString,
+      protected : protectedHeaderBase64UrlString,
+      signature : signatureBase64UrlString
+    };
+  }
+
+  /**
+   * Implementation of signing using ED25519.
+   */
+  private static async signEd25519 (
+    signingInputBuffer: Buffer,
+    privateKeyJwk: JwkEd25519Private
+  ): Promise<Buffer> {
+    const privateKeyBuffer = base64url.toBuffer(privateKeyJwk.d);
+    const signatureUint8Array = await ed25519.sign(signingInputBuffer, privateKeyBuffer);
+    const signatureBuffer = Buffer.from(signatureUint8Array);
+    return signatureBuffer;
+  }
+
+  /**
+   * Implementation of signing using SECP256K1.
+   */
+  private static async signSecp256k1 (
+    signingInputBuffer: Buffer,
+    privateKeyJwk: JwkSecp256k1Private
+  ): Promise<Buffer> {
+    const privateKeyBuffer = base64url.toBuffer(privateKeyJwk.d);
+    const signatureUint8Array = await secp256k1.sign(signingInputBuffer, privateKeyBuffer);
+    const signatureBuffer = Buffer.from(signatureUint8Array);
+    return signatureBuffer;
+  }
+
+  /**
+   * Verifies the JWS signature.
+   * @returns `true` if signature is successfully verified, false otherwise.
+   * @throws {Error} if key given is unsupported.
+   */
+  public static async verify (
+    jwsFlattenedModel: JwsFlattened,
+    publicKeyJwk: JwkPublic
+  ): Promise<boolean> {
+    const signatureInput = jwsFlattenedModel.protected + '.' + jwsFlattenedModel.payload;
+    const signatureInputBuffer = Buffer.from(signatureInput);
+    const signatureBuffer = base64url.toBuffer(jwsFlattenedModel.signature);
+
+    // This is where we will add support for different algorithms over time.
+    let result = false;
+    if (publicKeyJwk.crv === 'Ed25519') {
+      result = await Jws.verifyEd25519(signatureInputBuffer, signatureBuffer, publicKeyJwk as JwkEd25519Public);
+    } else if (publicKeyJwk.crv === 'secp256k1') {
+      result = await Jws.verifySecp256k1(signatureInputBuffer, signatureBuffer, publicKeyJwk as JwkSecp256k1Public);
```

Comment:
```suggestion
    } else if (publicKeyJwk.crv === 'P-256K') {
      result = await Jws.verifySecp256k1(signatureInputBuffer, signatureBuffer, publicKeyJwk);
```

the `as` cast can be removed because we used string literals for `kty` and `crv`